tracer/src/Datadog.Tracer.Native/clr_helpers.cpp

```diff
@@ -1018,7 +1018,318 @@ HRESULT HasAsyncStateMachineAttribute(const ComPtr<IMetaDataImport2>& metadataIm
+        TypeFromToken(resolutionScope) != mdtAssemblyRef && 
+        TypeFromToken(resolutionScope) != mdtModuleRef &&
+        resolutionScope != mdTokenNil)
+    {
+        tempToken = resolutionScope;
+        if (enclosingType == mdTokenNil)
+        {
+            enclosingType = tempToken;
+        }
+        hr = metadata_import->GetTypeRefProps(tempToken, &resolutionScope, nullptr, 0, &nameSize);
+        if (FAILED(hr))
+        {
+            Logger::Warn("[ResolveType] GetTypeRefProps [3] has failed. typeRefToken: ", typeRefToken);
+        }
+        IfFailRet(hr);
+    }
+
+    if (tryoutsCount == 0)
+    {
+        Logger::Warn(
+            "[ResolveType] Reached the maximum amount of tryouts of resolving the resolution scope. typeRefToken: ",
+            typeRefToken);
+        return E_FAIL;
+    }
+
+    if (resolutionScope == mdTokenNil)
+    {
+        Logger::Warn("[ResolveType] resolutionScope is nil. typeRefToken: ", typeRefToken);
+        return E_FAIL;
+    }
+
+    const auto& assemblyMetadata = GetReferencedAssemblyMetadata(assembly_import, resolutionScope);
+    if (assemblyMetadata.name == shared::EmptyWStr)
+    {
+        Logger::Warn("[ResolveType] GetReferencedAssemblyMetadata has failed. typeRefToken: ", typeRefToken);
+        return E_FAIL;
+    }
+
+    ICorProfilerModuleEnum* moduleEnum;
+    hr = info->EnumModules(&moduleEnum);
+    if (FAILED(hr))
+    {
+        Logger::Warn("[ResolveType] EnumModules has failed. typeRefToken: ", typeRefToken);
+    }
+    IfFailRet(hr);
+
+    std::vector<ModuleID> loadedModules;
+    size_t resultIndex = 0;
+
+    tryoutsCount = 1000;
+    // iterate over the loaded modules enumeration and collect the module ids into loadedModules
+    while (tryoutsCount-- > 0)
+    {
+        const ULONG valueToRetrieve = 20;
+        ULONG valueRetrieved = 0;
+
+        std::vector<ModuleID> tempValues(valueToRetrieve, 0);
+        hr = moduleEnum->Next(valueToRetrieve, tempValues.data(), &valueRetrieved);
+        if (FAILED(hr))
+        {
+            Logger::Warn("[ResolveType] EnumModules.Next has failed. typeRefToken: ", typeRefToken);
+        }
+        IfFailRet(hr);
+
+        if (valueRetrieved == 0)
+        {
+            break;
+        }
+
+        loadedModules.resize(loadedModules.size() + valueToRetrieve);
+        for (size_t k = 0; k < valueRetrieved; ++k)
+        {
+            loadedModules[resultIndex] = tempValues[k];
+            ++resultIndex;
+        }
+    }
+
+    if (tryoutsCount == 0)
+    {
+        Logger::Warn(
+            "[ResolveType] Reached the maximum amount of tryouts of enumerating the loaded modules. typeRefToken: ",
+            typeRefToken);
+        return E_FAIL;
+    }
+
+    const auto& resolutionScopeName = assemblyMetadata.name;
+
+    resolvedTypeDefToken = mdTokenNil;
+    if (enclosingType != mdTokenNil)
+    {
+        // LOG_DEBUG("ResolveType: Found enclosing type, try to get parent token");
```

Comment:
do you want to keep this debug comment?

Reply:
Revived this debug comment in ef502336ac5073c545e222bc6396635a9d76288f.